orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -98,10 +98,12 @@ def __init__(self, module, debug_options: DebugOptions, fallback_manager: _Fallb
         self._execution_agent = None
 
         # indicators of some logic have been executed previously thus could be skipped for faster training
+        # default is enabled, if not define in os env; but will be disabled with any exception
+        skip_check_enabled = _SkipCheck.SKIP_CHECK_DEVICE.name + '|' + _SkipCheck.SKIP_CHECK_BUILD_GRADIENT.name + '|' + _SkipCheck.SKIP_CHECK_EXECUTION_AGENT.name
```

Comment:
not directly related to this PR, but enabling/disabling using string parsing is not ideal. consider refactoring this piece

Reply:
discussed with Baiju and refactored